### PR TITLE
Add get server reload option

### PR DESCRIPTION
**Pull request by acurrieclark**
Not being particularly strong at the testing game, nor at git manipulation, I have not managed to squash my commits yet, or write tests which can detect whether a server has been loaded from the cache or not.
Any help or pointers with the tests in particular would be much appreciated. I will happily persevere with a little advice.

### CODE REVIEW

File: src/Forge.php [hunk 1]
```diff
@@ -135,16 +135,19 @@ public function create()
...
      * @return \Laravel\Forge\Server
      */
-    public function get(int $serverId)
+    public function get(int $serverId, bool $reload = null)
     {
-        if ($this->lazyLoadInitiated() && isset($this->serversMap[$serverId])) {
-            return $this->items[$this->serversMap[$serverId]];
-        } elseif (isset($this->serversCache[$serverId])) {
-            return $this->serversCache[$serverId];
+        if (!$reload) {
```
Comment (tzurbaev) [inline src/Forge.php]: I would rather check for `if ($reload === true)` and if `$reload` is actually true, just `return $this->loadSingleServer($serverId);`. This makes code look better since it uses only 1 indentation level.

```php
if ($reload === true) {
    return $this->loadSingleServer($serverId);
}

// Here goes original lazy load checks and servers cache returns.
```

File: src/Forge.php [hunk 2]
```diff
@@ -135,16 +135,19 @@ public function create()
...
      * Returns single server.
      *
-     * @param int $serverId
+     * @param int     $serverId
+     * @param boolean $reload   (optional) indicates whether the server should be reloaded
      *
      * @return \Laravel\Forge\Server
      */
-    public function get(int $serverId)
+    public function get(int $serverId, bool $reload = null)
```
Comment (tzurbaev) [inline src/Forge.php]: Default value for booleans should be either `true` or `false` (`false` in your case), but not `null`. Nulls are better suited for optional strings or objects (it depends, of course).

**Review by tzurbaev — CHANGES_REQUESTED** [review]

**Comment by tzurbaev** [comment]
Hi, @acurrieclark! Thank you for your pull request. I've added review comments with small requested changes. If you have any issues, please, reply to my comments in this review.

About testing: since this change may perform additional requests to Forge API at least one test should be added.

You can look at `ServersTest::testGetServer` method (`tests/ServersTest.php`) for more info. You can copy this test and add another API call expectation to the `Api::fake` closure.

Basically, you should load server first, then load it again (with `$reload = false`) and assert that the same server was returned from cache. And then you should reload it again with `$reload = true`. Second API call expectation should be called and test should be green.

If you having troubles with testing feel free to comment here.

**Comment by acurrieclark** [comment]
Hi @tzurbaev thanks for the feedback! Will get this fixed up.

File: src/Forge.php (hunk 2 above)
Comment (acurrieclark) [reply src/Forge.php]: Ah yes, this was a hangover from when I erroneously had the parameter erroneously typed as `boolean` rather than `bool`. I kept getting the `PHP Fatal error:  Default value for parameters with a class type can only be NULL` error. My mistake.

File: src/Forge.php (hunk 1 above)
Comment (acurrieclark) [reply src/Forge.php]: I have corrected this, but presumably I need to leave the default 
```php
return $this->loadSingleServer($serverId);
```
at the end of the function too?

Comment (tzurbaev) [reply src/Forge.php]: Yes, you're right.

**Comment by acurrieclark** [comment]
I am struggling to get a meaningful test sorted here. I can't work out how to test that a second http request to the same endpoint is made. Should I commit what I have so you can take a look?

**Comment by tzurbaev** [comment]
Don't worry, this is actually pretty easy. Look at this code:

```php
$api = Api::fake(function ($http) use ($data) {
    $http->shouldReceive('request')
        ->with('GET', 'servers/'.$data['id'])
        ->andReturn(
            FakeResponse::fake()->withJson(['server' => $data])->toResponse()
        );
});
```
I copied this from `ServersTest::testGetServer` method. As you can see, we're using HTTP Client mock here to ensure that we're not making actual calls to the Forge API. The `shouldReceive` method tells that `request` method should be called on `$http` object and this method call should return fake API response with given JSON structure.

You can simply add `->twice()` to the methods chain and this will assert that given method (`request`) was called exactly two times.

```php
$api = Api::fake(function ($http) use ($data) {
    $http->shouldReceive('request')
        ->with('GET', 'servers/'.$data['id'])
        ->twice()
        ->andReturn(
            FakeResponse::fake()->withJson(['server' => $data])->toResponse()
        );
});
```
Since calling `$forge->get('server-id', true)` will force `Forge` to send API request, the `request` method on `$http` object will be called again. And this should make your test green.

**Comment by acurrieclark** [comment]
Excellent and thank you. The only bit I was missing in my code was the `->twice()` section. Is that all guzzle mocking? Will get this submitted.

**Review by tzurbaev — APPROVED** [review]

**Comment by tzurbaev** [comment]
@acurrieclark great! Thank you for contributing! I'll tag these changes as v1.2.0 and release it soon.

**Comment by acurrieclark** [comment]
You are most welcome. Thanks for the tips
